common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsFileHelper.kt

```diff
@@ -0,0 +1,42 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import com.revenuecat.purchases.common.FileHelper
+import com.revenuecat.purchases.common.verboseLog
+import org.json.JSONObject
+
+/**
+ * All methods in this file should be executed within the diagnostics thread to ensure there are no threading issues.
+ */
+class DiagnosticsFileHelper(
```

Comment:
it'd be so nice to be able to use actors here

Reply:
I think so yeah. And yeah, we don't have that in Android 😢